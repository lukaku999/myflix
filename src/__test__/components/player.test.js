import React from 'react';
import {render, fireEvent} from '@testing-library/react'
import {Player} from '../../components/player/index'

describe('<Player/>', () => {
    it('render the <Player/> wth a bunny video', () => {

        const {container, getByText, queryByTestId} = render(
            <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4"/>
            </Player>
        )
        console.log(container)
        expect(queryByTestId('player')).toBeFalsy()
        fireEvent.click(getByText('Play'))

        expect(queryByTestId('player')).toBeTruthy()
        fireEvent.click(queryByTestId('Play'))

        expect(queryByTestId('player')).toBeFalsy()
        expect(container.firstChild).toMatchSnapshot()
    })
})