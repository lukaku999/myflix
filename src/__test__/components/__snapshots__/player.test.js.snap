// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> render the <Player/> wth a bunny video 1`] = `
<div
  class="player__Container-sc-1c49yiu-0"
>
  <button
    class="player__Button-sc-1c49yiu-4 iERisg"
  >
    Play
  </button>
</div>
`;
